operations/src/main/java/edu/unc/lib/boxc/operations/impl/download/DownloadBulkService.java

```diff
@@ -60,7 +61,12 @@ private void zipFiles(WorkObject workObject, AccessGroupSet agentPrincipals, Pat
             }
 
             Map<String, Integer> duplicates = new HashMap<>();
+            int count = 0;
+            var loopLimit = getLoopLimit(memberObjects.size());
             for (ContentObject memberObject : memberObjects ) {
+                if (count == loopLimit) {
```

Comment:
after thinking about this a bit more, I think this can be changed to just
```
if (count == fileLimit)
```
Since the loop is never going to go past the end of the array. Also, if we start count at 1, then we can set the limit to 100 instead of 99, which I think would be clearer about the intent.